src/db/repos/user_repo.ts

```diff
@@ -2,22 +2,54 @@ import { EntityRepository, Repository } from "typeorm";
 import { UserEntity } from "../entity/user_entity";
 import { Request, Response } from "express";
 
-
 @EntityRepository(UserEntity)
-export class UserRepository extends Repository<UserEntity>{
-    async createUser(email: String, firstName: String, lastName: String, userId: String) {
-        await this.createQueryBuilder("users").insert().values({
-            user_id: userId,
-            user_first_name: firstName,
-            user_last_name: lastName,
-            user_email: email,
-            user_gender: ""
-        }).execute();
-    }
 
-    async updateDBEmail(email: String) {
-        await this.createQueryBuilder("users").insert().values({
-            user_email: email,
-        }).execute();
+export class UserRepository extends Repository<UserEntity> {
+  //Create new User
+  async createUser(
+    gender: String,
+    firstName: String,
+    lastName: String,
+    userId: String
+  ) {
+    await this.createQueryBuilder("users")
+      .insert()
+      .values({
+        user_id: userId,
+        user_first_name: firstName,
+        user_last_name: lastName,
+        user_gender: gender,
+      })
+      .execute();
+  }
+
+  // //Update email
+  // async upadateEmail(req: Request, res: Response) {
+  //   const { user_email, user_id } = req.body;
+  //   try {
+  //     let data = await this.createQueryBuilder("user")
+  //       .update(UserEntity)
+  //       .set({
+  //         user_email: user_email,
+  //       })
+  //       .where("user_id =:user_id", { user_id: user_id })
+  //       .execute();
+  //     return res.send(data);
+  //   } catch (error) {
+  //     res.send(error);
+  //   }
+  // }
+
+  async userDetails(req: Request, res: Response) {
```

Comment:
You have to find user using userid not with email and you have to fetch all the details like all linked account.